pygac/reader.py

```diff
@@ -116,38 +127,181 @@ def __init__(self, interpolate_coords=True, adjust_clock_drift=True,
+        # second use case “diamond diagrams”.
+        # Check if the data set name matches the pattern
+        LOG.debug("validate header")
+        try:
+            data_set_name = header['data_set_name'].decode()
+        except UnicodeDecodeError:
+            raise ReaderError('Not able to decode the data set name!')
+        if not cls.data_set_pattern.match(data_set_name):
+            raise ReaderError(
+                'Data set name "%s" does not match!' % data_set_name)
+
+    def _read_scanlines(self, buffer, count):
+        """Read the scanlines from the given buffer
+
+        Args:
+            buffer (bytes, bytearray): buffer to read from
+            count (int): number of expected scanlines
+        """
+        line_count = len(buffer) // self.scanline_type.itemsize
+        if line_count < count:
+            LOG.warning(
+                "Expected %d scan lines, but found %d!"
+                % (count, line_count))
+            warnings.warn("Unexpected number of scanlines!",
+                          category=RuntimeWarning)
+        self.scans = np.frombuffer(
+            buffer, dtype=self.scanline_type, count=line_count)
+
+    @classmethod
+    def can_read(cls, filename, fileobj=None):
+        """Read the GAC/LAC data.
+
+        Args:
+            filename (str): Path to GAC/LAC file
+            fileobj: An open file object to read from. (optional)
+
+        Retruns:
+            result (bool): True if the reader can read the input
+        """
+        if fileobj:
+            pos = fileobj.tell()
         try:
-            yield file_object
+            archive_header, header = cls.read_header(
+                filename, fileobj=fileobj)
+            result = True
+        except (ReaderError, ValueError) as exception:
+            LOG.debug("%s failed to read the file! %s"
+                      % (cls.__name__, repr(exception)))
+            result = False
         finally:
-            file_object.close()
+            if fileobj:
+                fileobj.seek(pos)
+        return result
+
+    @classmethod
+    def fromfile(cls, filename, fileobj=None):
+        """Create Reader from file (alternative constructor)
+
+        Args:
+            filename (str): Path to GAC/LAC file
+
+        Kwargs:
+            fileobj (file object): Open file object to read from
+
+        Note:
+            The fileobj is useful when dealing with tar archives,
+            where the filename is given by the tarinfo object, but
+            the extracted file object's property 'name' is set to
+            the filename of the archive.
+        """
+        instance = cls()
+        instance.read(filename, fileobj=fileobj)
+        return instance
+
+    def _get_calibrated_channels_uniform_shape(self):
+        """Prepare the channels as input for gac_io.save_gac"""
+        channels = self.get_calibrated_channels()
+        assert channels.shape[-1] == 6
```

Comment:
@mraspaud Is using assertions in the code a good practice or should this be checked by the unit tests instead?

Reply:
I don't have anything agains the assertion here, but yes the unit tests should test this case.

Reply:
I would like to keep the assert at this location, because the Reader class implements the default behavior which assumes this condition, but of cause this does not replace a unit test.